packages/firestore/src/remote/remote_store.ts

```diff
@@ -391,7 +404,13 @@ export class RemoteStore implements TargetMetadataProvider {
     ) {
       // There was an error on a target, don't wait for a consistent snapshot
       // to raise events
-      return this.handleTargetError(watchChange);
+      try {
+        await this.handleTargetError(watchChange);
+      } catch (e) {
+        logDebug(LOG_TAG, 'Failed to remove target: ' + e);
```

Comment:
Can we log the query associated with the target here?

Reply:
Unless we go out of our way, we don't have access to the query here. I logged the target ID instead. We should however already have this information from the Watch stream. 

FWIW, I didn't add this logging to begin with since this is a debug log. The other logs were error logs in response to user behavior. They are always visible, and often without context.